chapters/05-comparison.Rmd

```diff
@@ -0,0 +1,195 @@
+# Comparison {#comparison}
+
+```{r, setup, echo=FALSE, eval=TRUE}
+library('reticulate')
+use_condaenv('ba-thesis')
+```
+
+```{python, pythonSetup, echo = FALSE, eval=TRUE}
+import warnings
+warnings.filterwarnings('ignore')
+
+import os
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
+
+from tf_kde.benchmark import runner
+from matplotlib import pyplot as plt
+import pandas as pd
+
+distributions_to_evaluate = [
+    'Gaussian',
+    'Uniform',
+    'Bimodal',
+    'SkewedBimodal',
+    'Claw',
+    'AsymmetricDoubleClaw'
+]
+
+xlim = [-8, 8]
+
+unrestricted_runtimes = pd.read_pickle('./benchmark/cpu_macbook/unrestricted_runtimes.pkl')
+restricted_runtimes = pd.read_pickle('./benchmark/cpu_macbook/restricted_runtimes.pkl')
+
+unrestricted_estimations = pd.read_pickle('./benchmark/cpu_macbook/unrestricted_estimations.pkl')
+restricted_estimations = pd.read_pickle('./benchmark/cpu_macbook/restricted_estimations.pkl')
+
+```
+
+To show the efficiency and performance of the different kernel density estimation methods implemented with TensorFlow a benchmarking suite was developed. It consists of three parts: a collection of distributions to use, a collection of methods to compare and a runner module that implements helper methods to execute the methods to test against the different distributions and plot the generated datasets nicely.
+
+## Benchmark setup
+
+To compare the different implementations multiple popular test distributions mentioned in Wand et al.[@wand1994kernel] were used. A simple normal distribution, a simple uniform distribution, a bimodal distribution comprised of two normals, a skewed bimodal distribution, a claw distribution that has spikes and one called asymmetric double claw that has different sized spikes left and right. The data is sampled randomly from each test distribution.
```

Comment:
test distributions -> mention that these should be approximated by the KDE (just to be clear it's not the kernel)